### PR TITLE
only resolve roles for bindings that matter

### PR DESCRIPTION
**Pull request by deads2k**
Only resolve roles for bindings that matter, so that we limit the scope of referential integrity problems between bindings and roles.  Addresses the last comment in https://bugzilla.redhat.com/show_bug.cgi?id=1192310, but I'm going to see I can a separate defect for the different symptom.

@liggitt easy.

### CODE REVIEW

File: pkg/authorization/rulevalidation/find_rules.go [hunk 1]
```diff
@@ -62,17 +60,23 @@ func (a *DefaultRuleResolver) getPolicyBindings(ctx kapi.Context) ([]authorizati
...
+	if !exists {
+		return nil, errors.New("user missing from context")
+	}
+
 	ret := make([]authorizationapi.RoleBinding, 0, len(policyBindings))
 	for _, policyBinding := range policyBindings {
-		for _, value := range policyBinding.RoleBindings {
-			ret = append(ret, value)
+		for _, roleBinding := range policyBinding.RoleBindings {
+			if doesApplyToUser(roleBinding.Users, roleBinding.Groups, user) {
```
Comment (smarterclayton) [inline pkg/authorization/rulevalidation/find_rules.go]: Please rename this "appliesToUser()" - this wording makes my brain hurt a bit

Comment (deads2k) [inline pkg/authorization/rulevalidation/find_rules.go]: done

File: test/integration/authorization_test.go [hunk 2]
```diff
@@ -78,6 +78,44 @@ func TestRestrictedAccessForProjectAdmins(t *testing.T) {
...
+		Client:           openshiftClient,
+		Users:            []string{"anypassword:valerie"},
+	}
+	if err := addValerie.Run(); err != nil {
+		t.Errorf("unexpected error: %v", err)
+	}
+
+	if err = openshiftClient.Roles("master").Delete("view"); err != nil {
+		t.Errorf("unexpected error: %v", err)
+	}
```
Comment (liggitt) [inline test/integration/authorization_test.go]: maybe try to run addValerie again and make sure you get an error when the role is invalid?

**Comment by liggitt** [comment]
suggestion for additional test, LGTM otherwise

**Comment by deads2k** [comment]
@liggitt Can you re-review changes to find_rules.go?  Integration tests caught an insidious bug caused by our magic container (aka context).

**Comment by liggitt** [comment]
yay tests. LGTM

**Comment by deads2k** [comment]
[merge]

**Comment by deads2k** [comment]
opened https://github.com/openshift/origin/issues/1234 to tight method preconditions

**Pull request merged**